### PR TITLE
Updating with a new Compat section

### PR DESCRIPTION
**Pull request by ChantalJuntao**
Updated Compat section and fixed Deprecations. 

I'm not sure if this should be v0.1.3 or v0.2.0 since we didn't make any major changes, but v0.1.2 doesn't currently compile.

### CODE REVIEW

**Comment by timholy** [comment]
Normally a change like this would be 0.1.3 (so you did it right). However, RegisterPenalty 0.3 requires Julia 1.6, and so this will too. Any update to the minimum Julia version should get a minor version bump. So I would change this to 0.2.0 and then also set the minimum Julia [compat] to 1.6.

The other thing you'd want to do is update the `.github/workflows/CI.yml` file to run the tests on Julia 1.6.

**Comment by ChantalJuntao** [comment]
@timholy I'll fix the issue with the CoordinateTransformation version in RegisterPenalty. However, I just realized that the deprecations in the test might only exist in a more recent version of julia. Should I make those deprecations backwards compatible with Julia version? Or just launch a 0.2.0?

**Comment by timholy** [comment]
Let's require a minimum of Image 0.25 for this version bump. That doesn't force a minor bump: if you're not actually changing the exports of this package in any way, then there's nothing incompatible, so you don't need a breaking release.

The convention is that only the Julia version requires a minor bump. If we release 1.0 of our lab packages, then even a minor bump is not breaking (versions can be declared, e.g., `RegisterDeformation = "1"` and then any 1.x version is acceptable).

File: Project.toml [hunk 1]
```diff
@@ -17,6 +17,13 @@ RegisterUtilities = "d4862ba2-f42c-5aeb-af4f-96a8884a16c4"
 StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
+[compat]
```
Comment (timholy) [inline Project.toml:20]: Let's add the compat for the packages in the `General` registry too?

**Comment by ChantalJuntao** [comment]
So we maintain the range of julia versions that are usable here. For RegisterPenalty, do we need to change julia = 1.6 to >= 1.6? or add 1.7 as a compatible version since I'm currently 1.7 and it seems to work for me?

File: Project.toml (hunk 1 above)
Comment (timholy) [reply Project.toml:20]: No, just hard-code the ones that got resolved in your manifest. I.e., 
```
(RegisterFit) pkg> st
Project RegisterFit v0.1.3
Status `~/.julia/dev/RegisterFit/Project.toml`
  [46a7138f] CenterIndexedArrays v0.2.1
⌅ [150eb455] CoordinateTransformations v0.5.1
  [916415d5] Images v0.25.1
  [a98d9a8b] Interpolations v0.13.5
  [2774e3e8] NLsolve v4.5.1
  [429524aa] Optim v1.6.2
  [67712758] RegisterCore v0.2.4
  [464fa2a9] RegisterPenalty v0.3.0
  [d4862ba2] RegisterUtilities v0.1.0
⌅ [90137ffa] StaticArrays v0.12.5
  [37e2e46d] LinearAlgebra
  [10745b16] Statistics
```
and so I would use
```
Images = "0.25"
Interpolations = "0.13"
NLsolve = "4"
Optim = "1"
```
and so on. Notice that I'm not going beyond the first non-zero entry in major/minor/patch.

**Comment by timholy** [comment]
1.6 is a good choice for a minimum version since it's an LTS (Long Term Support). If it works on 1.7 it *should* work on 1.6, but you'll find out if you make sure 1.6 runs in CI.yml.

**Pull request merged**